integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt

```diff
@@ -354,6 +355,25 @@ internal constructor(
         }
     }
 
+    private fun toHttpMethod(method: String, internalLogger: InternalLogger): RumHttpMethod {
+        return when (method.uppercase(Locale.US)) {
```

Comment:
> [!IMPORTANT]
> The official [HTTP rfc](https://www.rfc-editor.org/rfc/rfc9110.html#section-9) mentions three additional known methods: `CONNECT`, `OPTIONS` and `TRACE`. 
> Although they are rarely used in mobile apps, maybe it's worth adding them as supported values (but this means also adding them to the json schema). Some customers are using the `OTPIONS` value (cf [telemetry](https://dd-rum-telemetry.datadoghq.com/logs?query=status%3Aerror%20service%3Add-sdk-android%20%22valid%20http%20method%22%20&agg_q=status%2Cservice&cols=host%2Cservice&index=%2A&messageDisplay=inline&refresh_mode=sliding&sort_m=%2C&sort_t=%2C&stream_sort=desc&top_n=10%2C10&top_o=top%2Ctop&viz=pattern&x_missing=true%2Ctrue&from_ts=1695721329642&to_ts=1696326129642&live=true))

> [!NOTE]
> On a side note, there's also a much [wider list of method extensions](https://www.iana.org/assignments/http-methods/http-methods.xhtml), which should be considered "part of HTTP, as long as they’re listed in the appropriate registries". It might be overkill to add those as well, though, and I can't find any of those being reported in the telemetry org.

Reply:
Completely agree. For now I would propose to keep things as they are, since the goal of this PR is just to change the API and we can deal with RUM schema in the future.